File: test/language/expressions/python/FieldTypeTest.py
import Expressions


class FieldTypeTest(Expressions.TestCase):
    def testBitSizeOfWithOptional(self):
        containedType = self.api.ContainedType(True)
        fieldTypeExpression = self.api.FieldTypeExpression(containedType, self.EXTRA_VALUE)
        self.assertEqual(self.COMPOUND_TYPE_EXPRESSION_BIT_SIZE_WITH_OPTIONAL, fieldTypeExpression.bitsizeof())

    def testBitSizeOfWithoutOptional(self):
        containedType = self.api.ContainedType(needs_extra_value_=False)
        fieldTypeExpression = self.api.FieldTypeExpression()
        fieldTypeExpression.contained_type = containedType
        self.assertEqual(
            self.COMPOUND_TYPE_EXPRESSION_BIT_SIZE_WITHOUT_OPTIONAL, fieldTypeExpression.bitsizeof()
        )

    COMPOUND_TYPE_EXPRESSION_BIT_SIZE_WITH_OPTIONAL = 4
    COMPOUND_TYPE_EXPRESSION_BIT_SIZE_WITHOUT_OPTIONAL = 1

    EXTRA_VALUE = 0x02

File: test/language/expressions/java/expressions/FieldTypeTest.java
package expressions;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;

import expressions.field_type.ContainedType;
import expressions.field_type.FieldTypeExpression;

public class FieldTypeTest
{
    @Test
    public void bitSizeOfWithOptional()
    {
        final ContainedType containedType = new ContainedType(true);
        final FieldTypeExpression fieldTypeExpression = new FieldTypeExpression(containedType, EXTRA_VALUE);

        assertEquals(COMPOUND_TYPE_EXPRESSION_BIT_SIZE_WITH_OPTIONAL, fieldTypeExpression.bitSizeOf());
    }

    @Test
    public void bitSizeOfWithoutOptional()
    {
        final ContainedType containedType = new ContainedType(false);
        final FieldTypeExpression fieldTypeExpression = new FieldTypeExpression();
        fieldTypeExpression.setContainedType(containedType);

        assertEquals(COMPOUND_TYPE_EXPRESSION_BIT_SIZE_WITHOUT_OPTIONAL, fieldTypeExpression.bitSizeOf());
    }

    private static final int COMPOUND_TYPE_EXPRESSION_BIT_SIZE_WITH_OPTIONAL = 4;
    private static final int COMPOUND_TYPE_EXPRESSION_BIT_SIZE_WITHOUT_OPTIONAL = 1;

    private static final byte EXTRA_VALUE = 0x02;
}
